samples/I2CBMP280.js

```diff
@@ -0,0 +1,244 @@
+        print("BMP280 not found! ChipId " + reader.toString('hex') !== "58");
+
+    readCoefficients();
+
+    var powerData = new Buffer([bmp280.BMP280_REGISTER_CONTROL, 0]);
+    powerData.writeUInt8(bmp280.BMP280_PRESS_OVER | bmp280.BMP280_TEMP_OVER |
+                         bmp280.BMP280_NORMAL_MODE, 1)
+
+    i2cDevice.write(bmp280.BMP280_ADDRESS, powerData);
+
+    powerData.writeUInt8(bmp280.BMP280_REGISTER_CONFIG, 0);
+    powerData.writeUInt8(bmp280.BMP280_STANDBY | bmp280.BMP280_FILTER |
+                         bmp280.BMP280_SPI_3W_DISABLE, 1);
+
+    i2cDevice.write(bmp280.BMP280_ADDRESS, powerData);
+
+    /* Setup Grove LCD */
+    clear();
+    // Set our preferences for the Grove LCD
+    var setup = glcd.GLCD_CMD_FUNCTION_SET | (glcd.GLCD_FS_ROWS_2 |
+                                              glcd.GLCD_FS_DOT_SIZE_LITTLE |
+                                              glcd.GLCD_FS_8BIT_MODE);
+    setupData.writeUInt8(0, 0);
+    setupData.writeUInt8(setup, 1);
+    i2cDevice.write(glcd.GROVE_LCD_DISPLAY_ADDR, setupData);
+
+    setup = glcd.GLCD_CMD_DISPLAY_SWITCH | (glcd.GLCD_DS_DISPLAY_ON |
+                                            glcd.GLCD_DS_CURSOR_ON |
+                                            glcd.GLCD_DS_BLINK_ON);
+    setupData.writeUInt8(setup, 1);
+    i2cDevice.write(glcd.GROVE_LCD_DISPLAY_ADDR, setupData);
+
+    resetCursor(0, 0);
+    writeWord("@Temperature is ", msgData);    
+}
+
+function readCoefficients() {
+    //These need to be read as a burst read in order to get accurate numbers
+    dig_TAll = i2cDevice.burstRead(bmp280.BMP280_ADDRESS, 24,
+                                   bmp280.BMP280_REGISTER_DIG_T1);
+    dig_T1 = dig_TAll.readUInt16LE();
+    dig_T2 = dig_TAll.readUInt16LE(2);
+    dig_T3 = dig_TAll.readUInt16LE(4);
+
+    //Convert these to signed
+    dig_T1 = uint16toInt16(dig_T1);
+    dig_T2 = uint16toInt16(dig_T2);
+    dig_T3 = uint16toInt16(dig_T3);
+}
+
+function readTemperature() {
+    var var1, var2;
+
+    // Read the raw temperature value.
+    var tempBuffer = i2cDevice.burstRead(bmp280.BMP280_ADDRESS, 4,
+                                         bmp280.BMP280_REGISTER_TEMPDATA);
+    // Get the number from the buffer object
+    var tempDec = tempBuffer.readUInt32BE();
+    //Select the first 20 bits of 32 bit integer and maintains the sign
+    tempDec >>>= 12;
+
+    //Convert raw temperature to Celsius
+    var1 = ((((tempDec >> 3) - (dig_T1 << 1))) * (dig_T2)) >> 11;
+    var2 = (((((tempDec >> 4) - (dig_T1)) * ((tempDec >> 4) - (dig_T1))) >> 12) *
+            (dig_T3)) >> 14;
+
+    t_fine = var1 + var2;
+    T = ((t_fine * 5 + 128) >> 8) / 100 ;
+
+    print ("Temp = " + T);
+    var dec = (T * 10) | 0;
+    dec = (dec - (T | 0) * 10) | 0;
+    var tempStr = "@" + (T | 0)  + '.' + dec + "C";
+
+    //Draw icon for direction temp has changed
+    resetCursor(0, 1);
+    if (prevTemp < T) {
+        writeWord("@+", changeData);
```

Comment:
btw, is this "@" the special character it needs for the grove lcd?  since it seems every string needs to start with the "@", so you might have writeWord() appends that character automatically, then you just call it with a regular string.